refactor: use slices.Contains to simplify code

keeghcet opened the pull request:
<!--- DELETE MARKDOWN COMMENTS BEFORE SUBMITTING PULL REQUEST. -->

<!--- Provide a descriptive summary of your changes in the Title above. -->

#### Type of change

<!--- What type of change? Pick one option and delete the others. -->


- Improvement (improvement to code, performance, etc)


#### Description

There is a [new function](https://pkg.go.dev/slices@go1.21.0#Contains) added in the go1.21 standard library, which can make the code more concise and easy to read.


<!--- Describe your changes in detail, including motivation. -->

#### Additional details

<!--- Additional implementation details or comments to reviewers. -->
<!--- Summarize how the pull request was tested (if not obvious from commit). -->

#### Related issues

<!--- Include a link to any associated issues, e.g. Github issue or approved rfc. -->

<!---
#### Release Note
If change impacts current users, uncomment Release Note heading and provide
release note text.
Also, copy release note text into the release specific /release_notes file.
-->

<!--
Checklist (DELETE AFTER READING):

- `Signed-off-by` added to commits (required for DCO check to pass)
- Tests have been added/updated (required for bug fixes and features)
- Unit and/or integration tests pass locally
- Run linters and checks locally using 'make checks'
- If change requires documentation updates, make updates in pull request,
  or open a separate issue and provide link
- Squash commits into a single commit, unless a stack of commits is
  intentional to assist reviewers or to preserve review comments.
- For additional contribution guidelines see the project's CONTRIBUTING.md file
-->

keeghcet committed:
refactor: use slices.Contains to simplify code

Signed-off-by: keeghcet <keeghcet@outlook.com>

pfi79 commented on core/container/externalbuilder/externalbuilder.go:
```diff
@@ -418,10 +419,5 @@ func appendDefaultPropagateEnvironment(propagateEnvironment []string) []string {
-	}
-	return false
+	return slices.Contains(propagateEnvironment, key)
 }
```
it is necessary not to change the body of the contains function, but the calls themselves. the contains function is no longer needed.

keeghcet replied:
Thanks. Modifid!

pfi79 commented on core/container/externalbuilder/externalbuilder.go:
```diff
@@ -418,10 +419,5 @@ func appendDefaultPropagateEnvironment(propagateEnvironment []string) []string {
-	}
-	return false
+	return slices.Contains(propagateEnvironment, key)
 }
```
it is necessary not to change the body of the contains function, but the calls themselves. the contains function is no longer needed.

pfi79 commented on core/ledger/kvledger/pause_resume_test.go:
```diff
@@ -133,10 +134,5 @@ func assertLedgerStatus(t *testing.T, provider *Provider, genesisBlocks []*commo
-	}
-	return false
+	return slices.Contains(slice, val)
 }
```
it is necessary not to change the body of the contains function, but the calls themselves. the contains function is no longer needed.

keeghcet replied:
Done

pfi79 commented on core/ledger/kvledger/txmgmt/statedb/commontests/test_common.go:
```diff
@@ -1163,12 +1164,7 @@ func CreateTestData(t *testing.T, db statedb.VersionedDB, ns string, numKeys int
-	}
-	return false
+	return slices.Contains(s, str)
 }
```
it is necessary not to change the body of the contains function, but the calls themselves. the contains function is no longer needed.

keeghcet replied:
Done

pfi79 commented on core/ledger/kvledger/tests/env.go:
```diff
@@ -318,10 +319,8 @@ type membershipInfoProvider struct {
+		return true, nil
 	}
 	return false, nil
 }
```
in this function, the second parameter returns error, which is always nil. Why not get rid of this second parameter? It seems to me that this will seriously simplify the code.

keeghcet replied:
Thanks. I moidified.  Please review it again.

pfi79 reviewed (changes requested):
Technically, you did the right thing, but in my opinion, it can be improved. Try to do better.

keeghcet commented:
> Technically, you did the right thing, but in my opinion, it can be improved. Try to do better.

Thanks for your review!  Modified!

pfi79 commented:
> > Technically, you did the right thing, but in my opinion, it can be improved. Try to do better.
> 
> Thanks for your review! Modified!

please achieve a green pipline

keeghcet commented:
> > > Technically, you did the right thing, but in my opinion, it can be improved. Try to do better.
> > 
> > 
> > Thanks for your review! Modified!
> 
> please achieve a green pipline

Thanks. Modified.

keeghcet commented:
@pfi79  I have fix the mock test file, Please approve the CI and review again when you have time.


<img width="309" height="82" alt="image" src="https://github.com/user-attachments/assets/d5118676-b076-41d6-bae7-f9bd94be6dbd" />

pfi79 commented on core/ledger/mock/membership_info_provider.go:
have you changed mock through the [counterfeiter regeneration](https://github.com/hyperledger/fabric/blob/main/core/ledger/ledger_interface.go#L835)?

keeghcet replied:
Executed. Some warning but no error:

```shell
fabric git:(main) ✗ go generate ./core/ledger/ledger_interface.go

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `StateListener` to `mock/state_listener.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `QueryExecutor` to `mock/query_executor.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `TxSimulator` to `mock/tx_simulator.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `DeployedChaincodeInfoProvider` to `mock/deployed_ccinfo_provider.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `MembershipInfoProvider` to `mock/membership_info_provider.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `HealthCheckRegistry` to `mock/health_check_registry.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `ChaincodeLifecycleEventListener` to `mock/cc_event_listener.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `CustomTxProcessor` to `mock/custom_tx_processor.go`... Done

WARNING: Invoking counterfeiter multiple times from "go generate" is slow.
Consider using counterfeiter:generate directives to speed things up.
See https://github.com/maxbrunsfeld/counterfeiter#step-2b---add-counterfeitergenerate-directives for more information.
Set the "COUNTERFEITER_NO_GENERATE_WARNING" environment variable to suppress this message.

Writing `ChaincodeLifecycleEventProvider` to `mock/cc_event_provider.go`... Done
```

pfi79 replied:
I am aware of these warnings. This is normal

Your change concerns only the membership_info_provider file. Return the remaining mock files back. 

Just in case, I'll say that the tools directory indicates which version of the counterfeiter we are currently using.

keeghcet replied:
Okay, thank you for your guidance. I have reverted the other mock files.

pfi79 commented:
@keeghcet Look at the tests. It seems that some kind of test was touched. The test dropped twice in one place.

keeghcet commented:
> @keeghcet Look at the tests. It seems that some kind of test was touched. The test dropped twice in one place.

It seems that the assertion for the error should be nil because the error return value has been removed.


```shell
--- FAIL: TestSnapshotImporterErrorPropagation (0.09s)
    --- FAIL: TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error (0.01s)
        snapshot_data_importer_test.go:512: 
            	Error Trace:	/home/runner/work/fabric/fabric/core/ledger/pvtdatastorage/snapshot_data_importer_test.go:512
            	Error:      	An error is expected but got nil.
            	Test:       	TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error
```

Modified.

pfi79 commented:
> > @keeghcet Look at the tests. It seems that some kind of test was touched. The test dropped twice in one place.
> 
> It seems that the assertion for the error should be nil because the error return value has been removed.
> 
> ```shell
> --- FAIL: TestSnapshotImporterErrorPropagation (0.09s)
>     --- FAIL: TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error (0.01s)
>         snapshot_data_importer_test.go:512: 
>             	Error Trace:	/home/runner/work/fabric/fabric/core/ledger/pvtdatastorage/snapshot_data_importer_test.go:512
>             	Error:      	An error is expected but got nil.
>             	Test:       	TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error
> ```
> 
> Modified.

Take a look at this test `error-when-membershipProvider-returns-error`. He doesn't match the description right now. We need to do something about it. It may need to be deleted.

keeghcet commented:
> > > @keeghcet Look at the tests. It seems that some kind of test was touched. The test dropped twice in one place.
> > 
> > 
> > It seems that the assertion for the error should be nil because the error return value has been removed.
> > ```shell
> > --- FAIL: TestSnapshotImporterErrorPropagation (0.09s)
> >     --- FAIL: TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error (0.01s)
> >         snapshot_data_importer_test.go:512: 
> >             	Error Trace:	/home/runner/work/fabric/fabric/core/ledger/pvtdatastorage/snapshot_data_importer_test.go:512
> >             	Error:      	An error is expected but got nil.
> >             	Test:       	TestSnapshotImporterErrorPropagation/error-when-membershipProvider-returns-error
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Modified.
> 
> Take a look at this test `error-when-membershipProvider-returns-error`. He doesn't match the description right now. We need to do something about it. It may need to be deleted.

Thanks for your help. Removed!

C0rWin commented:
@keeghcet hey, thanks for your contribution, do you have any plans to finalize the PR?
